ex.py

```diff
@@ -0,0 +1,31 @@
+import functools
+
+
+class TypeErrorWithMessage(TypeError):

```

Comment:
Its bad practice to build a dummy class without implementing it.
Read about how to implement a custom error class in a good way.